text/0000-on-modifier.md

```diff
@@ -0,0 +1,196 @@
+- They are not compatible at all with standard web components. Web components
+  don't send events in the same way as standard elements for properties, and
+  instead rely on event listeners added via `addEventListener`. While web
+  components are not currently used in the Ember ecosystem, they are advancing
+  and may eventually be useful for smaller, self-contained components,
+  especially for users that want to share code between framework ecosystems.
+
+Neither of these solutions is perfect for solving the problem of adding event
+listeners. This RFC proposes adding a new modifier to Ember: the `{{on}}`
+modifier. This modifier will explicitly add event listeners using the
+`addEventListener` API.
+
+## Credits
+
+The design of this modifier is based on [ember-on-modifier](https://github.com/buschtoens/ember-on-modifier)
+by Jan Buschtöns, which is an excellent addon that has allowed us to test this
+design in real apps and get feedback about the design.
+
+## Detailed design
+
+The `{{on}}` modifier will recieve:
+
+1. The event name as a string as the first positional parameter
+2. The event listener function as the second positional parameter
+3. Named parameters as options for the event listener, such as `capture`,
+   `once`, and `passive`
+
+The following usages are equivalent:
+
+```hbs
+<div {{on "click" this.handleClick passive=true}}></div>
+```
+
+```js
+element.addEventListener('click', this.handleClick, { passive: true });
+```
+
+Multiple event listeners can be added to an element by adding the modifier more
+than once:
+
+```hbs
+<div
+  {{on "click" this.handleClick}}
+  {{on "mouseenter" this.handleMouseEnter}}
+>
+</div>
+```
+
+Extra positional parameters will be ignored. In order to pass values to a
+function, users should use a helper such as the [bind helper](https://github.com/emberjs/rfcs/pull/470):
+
+```hbs
+<div {{on "click" (bind this.addNumber 123)}}></div>
+```
+
+The event will also be passed as a parameter to the function, so it can be used
+directly.
+
+`{{on}}` does _not_ bind context. Context binding must be handled via a helper,
+or in the component definition via the `@action` decorator, which will be the
+recommended path:
+
+```js
+class ClickComponent extends Component {
+  @action
+  handleClick() {
+    // ...
+  }
+}
+```
+
+## How we teach this
+
+`{{on}}` itself can be taught in conjuction with event listeners. We can give
+users an overview of event listeners and the `addEventListener` API, and then
+relate the modifier directly to it. We should provide lots of examples of usage
+throughout the guides for how users can use it, including some examples of how
+to apply multiple event listeners to an element.
+
+`{{on}}` also does not bind context, and we should make it clear that the
+`@action` decorator is the recommended way to bind methods to their components.
+
+Other than that, most examples in the guides can be translated directly from
+the `{{action}}` modifier to the `{{on}}` modifier.
+
```

Comment:
```suggestion
Documentation work would include:

- revising [Handling Events](https://guides.emberjs.com/release/components/handling-events/#toc_event-names) corresponding Octane article in entirety
- revising [Triggering Changes with Actions](https://guides.emberjs.com/release/components/triggering-changes-with-actions/)'s corresponding Octane article
- Updating 50+ uses of `{{action}}` in the guides
- Adding `{{on}}` as a new API docs entry 
- Update examples within the API docs to show all uses of actions that are supported in the API docs - `{{on}}`, `{{action}}`, `onclick={{action}}`

`{{action}}` is used 130+ times in the API docs. Not all examples would need to be updated, however an audit would be needed to figure out which need to be added to/changed.
```

Reply:
For context, the numbers above come from searching `guides-source/guides/release/*` for `{{action`, and in the API docs, running `yuidoc .` and searching the resulting `docs/data.json`

Reply:
Yeah updating docs for this and #470 would be a big lift (though fairly straightforward hopefully).

Some of the examples of `action` may be either `on` or `bind` depending on modifier or helper usage.